src/Builder.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations;
+
+use Doctrine\Annotations\Annotation\Target;
+use Doctrine\Annotations\Metadata\ClassMetadata;
+use Doctrine\Annotations\Metadata\MetadataFactory;
+use Doctrine\Annotations\Exception\TypeMismatchException;
+use Doctrine\Annotations\Exception\TargetNotAllowedException;
+use Doctrine\Annotations\Exception\InvalidAnnotationException;
+
+/**
+ * Build annotations objects
+ *
+ * @author Fabio B. Silva <fabio.bat.silva@gmail.com>
+ */
+class Builder
+{
+    /**
+     * @var \Doctrine\Annotations\Resolver
+     */
+    private $resolver;
+
+    /**
+     * @var \Doctrine\Annotations\MetadataFactory
+     */
+    private $metadataFactory;
+
+    /**
+     * Constructor.
+     *
+     * @param \Doctrine\Annotations\Resolver        $resolver
+     * @param \Doctrine\Annotations\MetadataFactory $metadataFactory
```

Comment:
The type is incorrect, it should be `\Doctrine\Annotations\Metadata\MetadataFactory`.